Add information for the Recommended Versions (#566)
odidev/ecosystem-dashboard-for-arm: content/opensource_packages/openmpi.md

        partner_content:
        official_docs: https://docs.open-mpi.org/en/v5.0.x/installing-open-mpi/quickstart.html
    arm_recommended_minimum_version:
        version_number: 5.0.4
        release_date: 2024/07/19
        reference_content: https://docs.open-mpi.org/en/v5.0.x/release-notes/changelog/v5.0.x.html#open-mpi-version-v5-0-4

>         rationale: In this version, support for Aarch64 OP component for for NEON and SVE ISA were added.

Author: ```suggestion
        rationale: In this release, support was added for the Aarch64 OP component for the NEON and SVE ISAs.
```
